pymc/gp/hsgp_approx.py

```diff
@@ -602,21 +606,27 @@ def prior(self, name: str, X: TensorLike, dims: Optional[str] = None):  # type:
         (phi_cos, phi_sin), psd = self.prior_linearized(X - self._X_mean)
 
         m = self._m
-        self._beta = pm.Normal(f"{name}_hsgp_coeffs_", size=(m * 2 - 1))
-        # The first eigenfunction for the sine component is zero
-        # and so does not contribute to the approximation.
-        f = (
-            self.mean_func(X)
-            + phi_cos @ (psd * self._beta[:m])  # type: ignore
-            + phi_sin[..., 1:] @ (psd[1:] * self._beta[m:])  # type: ignore
-        )
 
-        self.f = pm.Deterministic(name, f, dims=dims)
+        if self.drop_first:
```

Comment:
I wonder is we would like to inform the user when `drop_first=Flase` and the fist basis vectors are zero or ones

Reply:
Agree this would be very cool.  The basis vectors are made in pytensor.  I tried using `constant_fold` to automatically check if the first sin term is all zeros and drop it automatically, but since X can be mutable it didn't work.